src/plugins/list.js

```diff
@@ -11,7 +13,35 @@ function List() {
    * @param {Editor} editor
    * @param {Function} next
    */
-  const onEnter = (event, editor, next) => next();
+  const onEnter = (event, editor, next) => {
+    const { value } = editor;
+    const { startBlock } = value;
+    event.preventDefault();
+
+    // Hitting enter on a blank list item will break out of the enclosing list
+    if (isSelectionList(value) && startBlock.text.length === 0) {
+      editor.withoutNormalizing(() => {
+        event.preventDefault();
+        editor
+          .setBlocks(CONST.PARAGRAPH)
+          .unwrapBlock(CONST.LIST_ITEM)
+          .unwrapBlock(CONST.OL_LIST)
```

Comment:
I do not understand the double unwrapping of first `OL_LIST` and then `UL_LIST`. Shouldn't this unwrap only one? (i.e., the first enclosing list?)

Reply:
Good point. I think Slate was handling this as one unwrap to whichever list it was in. I refactored to unwrap once for whichever list it was.